cmd/kops/create_secret_cacert.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	"k8s.io/kops/cmd/kops/util"
+	"k8s.io/kops/pkg/pki"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/utils"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	"k8s.io/kubernetes/pkg/kubectl/util/i18n"
+)
+
+var (
+	createSecretCacertLong = templates.LongDesc(i18n.T(`
+	Add a ca certificate and private key.
+    `))
+
+	createSecretCacertExample = templates.Examples(i18n.T(`
+	Add a ca certificate and private key.
+	kops create secret ca-cert \
```

Comment:
I believe we're creating a keypair, in particular the CA keypair.  So we could do `kops create secret keypair ca --cert ~/ca.pem --key ~/ca-key.pem ...`.  (You'll note that CertificateId_CA == ca , happily enough :-) )

That would then let us create any keypair in the system, which could be really nice, particularly for serving with alternative keypairs on the apiserver, which is a common request.  (And if you do `kops get secret`, you'll see that we do call them keypairs there also).

I could see an argument that we should special-case the CA keypair, but on the other hand it feels like the `kops create secret keypair <name>` is much more general.  What do you think?

There's also a formal notion of keysets coming in the API - a keyset being multiple keypair versions.  But the API for that wouldn't be as nice as what you've got here, I don't think.

Reply:
that makes since to do. Did some changes to have a new sub command group to reflect what you wrote